File: app/src/main/java/com/sasfmlzr/findfilm/request/DiscoverMovieRequest.java
package com.sasfmlzr.findfilm.request;

import android.graphics.Bitmap;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@SuppressWarnings({"unused"})
public class DiscoverMovieRequest {
    @SerializedName("page")
    @Expose
    private int page;
    @SerializedName("results")
    @Expose
    private List<Result> results = null;
    @SerializedName("total_results")
    @Expose
    private int totalResults;
    @SerializedName("total_pages")
    @Expose
    private int totalPages;

    public DiscoverMovieRequest() {
    }

    public DiscoverMovieRequest(int page, List<Result> results, int totalResults, int totalPages) {
        super();
        this.page = page;
        this.results = results;
        this.totalResults = totalResults;
        this.totalPages = totalPages;
    }

    public int getPage() {
        return page;
    }

    public List<Result> getResults() {
        return results;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public static class Result {

        @SerializedName("poster_path")
        @Expose
        private Object posterPath;
        @SerializedName("adult")
        @Expose
        private boolean adult;
        @SerializedName("overview")
        @Expose
        private String overview;
        @SerializedName("release_date")
        @Expose
        private String releaseDate;
        @SerializedName("genre_ids")
        @Expose
        private List<Integer> genreIds = null;
        @SerializedName("id")
        @Expose
        private int id;
        @SerializedName("original_title")
        @Expose
        private String originalTitle;
        @SerializedName("original_language")
        @Expose
        private String originalLanguage;
        @SerializedName("title")
        @Expose
        private String title;
        @SerializedName("backdrop_path")
        @Expose
        private Object backdropPath;
        @SerializedName("popularity")
        @Expose
        private double popularity;
        @SerializedName("vote_count")
        @Expose
        private int voteCount;
        @SerializedName("video")
        @Expose
        private boolean video;
        @SerializedName("vote_average")
        @Expose
        private double voteAverage;
        private Bitmap backdropBitmap;

        public Result() {
        }

        public Result(Object posterPath, boolean adult, String overview, String releaseDate, List<Integer> genreIds, int id, String originalTitle, String originalLanguage, String title, Object backdropPath, double popularity, int voteCount, boolean video, double voteAverage) {
            super();
            this.posterPath = posterPath;
            this.adult = adult;
            this.overview = overview;
            this.releaseDate = releaseDate;
            this.genreIds = genreIds;
            this.id = id;
            this.originalTitle = originalTitle;
            this.originalLanguage = originalLanguage;
            this.title = title;
            this.backdropPath = backdropPath;
            this.popularity = popularity;
            this.voteCount = voteCount;
            this.video = video;
            this.voteAverage = voteAverage;
        }

        public Object getPosterPath() {
            return posterPath;
        }

        public boolean isAdult() {
            return adult;
        }

        public String getOverview() {
            return overview;
        }

        public String getReleaseDate() {
            return releaseDate;
        }

        public List<Integer> getGenreIds() {
            return genreIds;
        }

        public int getId() {
            return id;
        }

        public String getOriginalTitle() {
            return originalTitle;
        }

        public String getOriginalLanguage() {
            return originalLanguage;
        }

        public String getTitle() {
            return title;
        }

        public Object getBackdropPath() {
            return backdropPath;
        }

        public double getPopularity() {
            return popularity;
        }

        public int getVoteCount() {
            return voteCount;
        }

        public boolean isVideo() {
            return video;
        }

        public double getVoteAverage() {
            return voteAverage;
        }

        public Bitmap getBackdropBitmap() {
            return backdropBitmap;
        }

        public void setBackdropBitmap(Bitmap backdropBitmap) {
            this.backdropBitmap = backdropBitmap;
        }
    }
}

File: app/src/main/java/com/sasfmlzr/findfilm/request/DiscoverMovieRequest.kt
package com.sasfmlzr.findfilm.request

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class DiscoverMovieRequest {
    @SerializedName("page")
    @Expose
    var page = 0
        private set

    @SerializedName("results")
    @Expose
    var results: List<Result>? = null
        private set

    @SerializedName("total_results")
    @Expose
    var totalResults = 0
        private set

    @SerializedName("total_pages")
    @Expose
    var totalPages = 0
        private set

    constructor() {}
    constructor(page: Int, results: List<Result>?, totalResults: Int, totalPages: Int) : super() {
        this.page = page
        this.results = results
        this.totalResults = totalResults
        this.totalPages = totalPages
    }

    class Result {
        @SerializedName("poster_path")
        @Expose
        var posterPath: Any? = null
            private set

        @SerializedName("adult")
        @Expose
        var isAdult = false
            private set

        @SerializedName("overview")
        @Expose
        var overview: String? = null
            private set

        @SerializedName("release_date")
        @Expose
        var releaseDate: String? = null
            private set

        @SerializedName("genre_ids")
        @Expose
        var genreIds: List<Int>? = null
            private set

        @SerializedName("id")
        @Expose
        var id = 0
            private set

        @SerializedName("original_title")
        @Expose
        var originalTitle: String? = null
            private set

        @SerializedName("original_language")
        @Expose
        var originalLanguage: String? = null
            private set

        @SerializedName("title")
        @Expose
        var title: String? = null
            private set

        @SerializedName("backdrop_path")
        @Expose
        var backdropPath: Any? = null
            private set

        @SerializedName("popularity")
        @Expose
        var popularity = 0.0
            private set

        @SerializedName("vote_count")
        @Expose
        var voteCount = 0
            private set

        @SerializedName("video")
        @Expose
        var isVideo = false
            private set

        @SerializedName("vote_average")
        @Expose
        var voteAverage = 0.0
            private set

        constructor() {}
        constructor(
            posterPath: Any?,
            adult: Boolean,
            overview: String?,
            releaseDate: String?,
            genreIds: List<Int>?,
            id: Int,
            originalTitle: String?,
            originalLanguage: String?,
            title: String?,
            backdropPath: Any?,
            popularity: Double,
            voteCount: Int,
            video: Boolean,
            voteAverage: Double
        ) : super() {
            this.posterPath = posterPath
            isAdult = adult
            this.overview = overview
            this.releaseDate = releaseDate
            this.genreIds = genreIds
            this.id = id
            this.originalTitle = originalTitle
            this.originalLanguage = originalLanguage
            this.title = title
            this.backdropPath = backdropPath
            this.popularity = popularity
            this.voteCount = voteCount
            isVideo = video
            this.voteAverage = voteAverage
        }
    }
}